BCStudentSoftwareDevTeam/lsf
# Introduce SupervisorDepartment Table to Allow Supervisors to add Other Supervisors to Their Departments

Fixes issue #365 
## Changes Made
- Created the new model SupervisorDepartment.
- Added a button to the Supervisor Portal that allows a supervisor to add another supervisor to their department.
- Adjusted the logic function getDepartmentsForSupervisor to take into account this new table.
- Changed the tests to reflect our changes.
- If only one department exists for a supervisor, default select that department on Form Search.

## Testing
- `source setup.sh`
- `./database/reset_database.sh`
- `./tests/run_tests.sh`
- `./database/reset_database.sh from-backup`
- `export FLASK_ENV=staging`
- `flask run`
- Change your user to `heggens`
- On the supervisorportal page, choose a user to add to a department, we used `cupidonj` who only is a part of Mathematics and Economics
- Change your user to the user you added to a department and make sure you can now interact with that department.
- Add the user to a department that they already have forms in (for `cupidonj` it would be either Mathematics or Economics) and ensure no duplicates are created.
